[OG-126] projects frontend

kligarski opened the pull request:
Adds creating, editing and deleting project groups on the frontend.

https://github.com/user-attachments/assets/3fbd3f0f-2e24-4f63-beb1-74e1da33d3fd

https://github.com/user-attachments/assets/39b528f2-1b94-42c4-8a21-e6528f66c599

https://github.com/user-attachments/assets/0caf97b6-2950-4a69-a615-0dade25317c4

TODO:
- ~~RWD~~
- ~~figure out if we can avoid 2 modals at the same time~~ fixed, there is a little flash but I don't think that we can really do anything about it.
- ~~code cleanup~~

kligarski committed:
skeleton for project group configuration modal

kligarski committed:
context base

kligarski committed:
group pick

kligarski committed:
remove old group pick

kligarski committed:
project variant base

kligarski committed:
add filters

kligarski committed:
accordion fix + variant picking v1

kligarski committed:
project group configuration submission

kligarski committed:
project group deletion

kligarski committed:
fix modal import

krudny commented on polymorphia-frontend/providers/grading/types.ts:
```diff
@@ -9,15 +9,15 @@ import {
   criteria: CriterionResponseDTO[] | undefined;
   submissionRequirements: SubmissionRequirementResponseDTO[] | undefined;
-  projectVariants: ProjectVariantResponseDTO[] | undefined;
+  projectVariants: ProjectVariantWithCategoryNameResponseDTO[] | undefined;
```
maybe we should use ? operator there? idk, we are inconsistent everywhere with that

kligarski replied:
https://github.com/krudny/Polymorphia/pull/248/commits/b74800c243603ce72658dfb7b81a0a618a39f16a

kligarski committed:
rwd

kligarski committed:
prevent overlapping modals

kligarski committed:
css cleanup + types

kligarski committed:
small cleanup

kligarski committed:
absolute imports

krudny commented:
Apart from me, no one will see that, but we have inconsistency in scrollbar positions.

I like how it is done in hof: 

<img width="421" height="273" alt="image" src="https://github.com/user-attachments/assets/3f7ed9fd-d0f3-46ba-8cb5-975722c82025" />

Grading (XD):  

<img width="460" height="493" alt="image" src="https://github.com/user-attachments/assets/0ec07704-bbae-43ad-a6ab-93ccfae7f4b0" />


Grading mobile (heart broken cat jpg):

<img width="514" height="344" alt="image" src="https://github.com/user-attachments/assets/5227c6be-7088-493b-8812-e832b6e12870" />

It is not important at all, could be an issue  

I scrolled through the code, looks as working and great. I dont know when i will have time to do proper review, but I will try to find some.

krudny commented on polymorphia-frontend/components/speed-dial/modals/project-group-configuration/group-pick/index.css:
```diff
@@ -0,0 +1,38 @@
+}
+
+.group-pick-filters-button {
+  @apply flex gap-x-2 lg:gap-x-4 mt-1;
```
breakpoint styles should be last, and sorted from smallest

kligarski replied:
https://github.com/krudny/Polymorphia/pull/248/commits/ef462daff2a54e6714e44042c6ff8b19c6919492

kligarski committed:
add button to randomize project variants

kligarski committed:
sort tailwind classes in .css

krudny reviewed (changes requested):
Please provide reliable data. I checked develop and prod db, none of them work (target lists are not loading). 

https://drive.google.com/file/d/1oFUd1fP1DXvJoUSQOVDfEDVC9RIhIWWL/view?usp=sharing

I have no idea how I was able to get to this, I cant reproduce. Maybe it is db inconsistency error, but entering into projects, I have some target lists and git event section, and RWD completely crashed (i am not able to scroll grading components as expected). After that, clicking once again to project, i have the same url, but no targets xD

Do you see this animation lag, when I enter git grading? i think it is because of devtools are opened and a lot of rerendering is going on, but i am not sure if next js turbopack has something with it, or chrome devtools are so bad they couldnt run our animations. 

I would highly recommend to make `example.sql` great (or usable) again. 

<img width="644" height="185" alt="image" src="https://github.com/user-attachments/assets/f421e78e-1fcd-4b10-9e48-dd9878e14b31" />
 
from ss you have provided, roundings not matching

kligarski commented:
> I have no idea how I was able to get to this, I cant reproduce. Maybe it is db inconsistency error, but entering into projects, I have some target lists and git event section, and RWD completely crashed (i am not able to scroll grading components as expected). After that, clicking once again to project, i have the same url, but no targets xD

At the beginning, you have "project" in url for assignment section - that's why it breaks. Not sure how to get into this URL without modifying URL directly.

<img width="228" height="27" alt="obraz" src="https://github.com/user-attachments/assets/6948beb9-e16c-4740-985c-b08f4d58307a" />

<img width="253" height="29" alt="obraz" src="https://github.com/user-attachments/assets/c839f649-e415-4f06-9a20-81be786e2f04" />

> Please provide reliable data. I checked develop and prod db, none of them work (target lists are not loading).

> I would highly recommend to make example.sql great (or usable) again.

I think that the best would be to wait for projects backend.

> Do you see this animation lag, when I enter git grading? i think it is because of devtools are opened and a lot of rerendering is going on, but i am not sure if next js turbopack has something with it, or chrome devtools are so bad they couldnt run our animations.

Might be the devtools.

> from ss you have provided, roundings not matching

Not sure what you mean.

https://github.com/user-attachments/assets/39defd04-8537-4196-85ea-5970e6dcfcb6

kligarski committed:
block random button while mutation is pending

kligarski committed:
remove incorrect import

krudny commented:
please update branch!!!

10/10 vulnerability, vercel refuses to deploy that

kligarski committed:
bump lockfile

krudny commented:
<img width="444" height="185" alt="image" src="https://github.com/user-attachments/assets/e1096e57-3e96-4d7f-a8eb-bb04fe638ddc" />

not aligned

kligarski committed:
fix alignment

krudny commented:
poor rwd

<img width="377" height="669" alt="image" src="https://github.com/user-attachments/assets/8b41a091-abbd-4ac4-a6f0-cd8e96447f7c" />

krudny commented:
<img width="1195" height="504" alt="image" src="https://github.com/user-attachments/assets/a5b4bc68-ec1b-44a6-b202-81a265e60ebf" />

when is hidden and is locked are set to true, coordinator sees Projekt, instructor sees "*****" in ocenianie. That should be unified to one logical approach

krudny commented:
<img width="474" height="378" alt="image" src="https://github.com/user-attachments/assets/c68c44c3-d6fe-4ec0-a399-4366b8e0ce9c" />

should be exclusive, all or specific

krudny commented:
images for test data should be provided to me, i am using deep.ai to generate it

krudny commented:
<img width="294" height="132" alt="image" src="https://github.com/user-attachments/assets/9bfd51b4-68e5-4597-bc7c-20678605d50f" />

I would move [] content after category name to prevent layout shifts

kligarski commented:
> poor rwd
> <img alt="image" width="377" height="669" src="https://private-user-images.githubusercontent.com/134285490/526195186-8b41a091-abbd-4ac4-a6f0-cd8e96447f7c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU2NDExOTYsIm5iZiI6MTc2NTY0MDg5NiwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MTk1MTg2LThiNDFhMDkxLWFiYmQtNGFjNC1hNmYwLWNkOGU5NjQ0N2Y3Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxM1QxNTQ4MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yODNlNTYxZTE2NmVmOTI0OWZkNmNmNmY1MWUxNjhmYTdkYTg1NTM1ZGZmNmU5ZDY2ODViMDQ5NmZiYTdjMzU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Rs2sjBLMSY5b09k9-yBVkHlU2GInicr9rIimG54VHvs">

I'll have a look.

> <img alt="image" width="1195" height="504" src="https://private-user-images.githubusercontent.com/134285490/526195279-a5b4bc68-ec1b-44a6-b202-81a265e60ebf.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU2NDExOTYsIm5iZiI6MTc2NTY0MDg5NiwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MTk1Mjc5LWE1YjRiYzY4LWVjMWItNDRhNi1iMjAyLTgxYTI2NWU2MGViZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxM1QxNTQ4MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xNWM3ZDgxYzdiNTQzNjc2ZTVhOGJkYTUwMWU5YmQ0YWU5OGJmMjU4NGU2NDE2NTA2YmY2MDIwYjdmY2IwODk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Pd0xl9BxP3hV_S3RiLDClKGD7DzYpbX4ZiBgDkKiylE">
> 
> when is hidden and is locked are set to true, coordinator sees Projekt, instructor sees "*****" in ocenianie. That should be unified to one logical approach

This is handled by the backend.

> <img alt="image" width="474" height="378" src="https://private-user-images.githubusercontent.com/134285490/526195339-c68c44c3-d6fe-4ec0-a399-4366b8e0ce9c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU2NDExOTYsIm5iZiI6MTc2NTY0MDg5NiwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MTk1MzM5LWM2OGM0NGMzLWQ2ZmUtNGVjMC1hMzk5LTQzNjZiOGUwY2U5Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxM1QxNTQ4MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMTg5Mzg1Yjg1NTE5ZjQ0MDZjOWY5NDZmYjM2MWMzOGYxZjk3MDI3M2YzZmJhOGE3NzM1YTRiYzVmMzM2ZmY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2Qn91qpGFxks3LOfpqL2PPBshgxv8qCxSn4xOJh72YY">
> 
> should be exclusive, all or specific

This is handled by the backend.

> images for test data should be provided to me, i am using deep.ai to generate it

Should be handled with backend PR.

> <img alt="image" width="294" height="132" src="https://private-user-images.githubusercontent.com/134285490/526195497-9bfd51b4-68e5-4597-bc7c-20678605d50f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU2NDExOTYsIm5iZiI6MTc2NTY0MDg5NiwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MTk1NDk3LTliZmQ1MWI0LTY4ZTUtNDU5Ny1iYzdjLTIwNjc4NjA1ZDUwZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxM1QxNTQ4MTZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lYTdkZmQ3ZDcwY2U3ZmFhZGQxYjRlOTU4NzM1OTUzMDQ2NzNhNTI1MDE0YzdkZmJmMTJlYTVlOTg1MzlkY2NkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.e3ClhIGX4wRTu4ca5EKGXuO4Vp_goM4ZpgSTcxk2a_o">
> 
> I would move [] content after category name to prevent layout shifts

If the title gets longer, it'll be truncated with "...". Having the type at the beginning prevents that. But I can change it if you prefer having it at the end.

krudny commented:
>If the title gets longer, it'll be truncated with "...". Having the type at the beginning prevents that. But I can change it if you prefer having it at the end.

cant we shrink just category to always fit [] content?

krudny commented on polymorphia-frontend/providers/project-group-configuration/index.tsx:
```diff
@@ -0,0 +1,76 @@
+  );
+
+  const {
+    data: initialProjectGroupConfiugation,
```
typo

kligarski replied:
https://github.com/krudny/Polymorphia/pull/248/commits/b8efe44dc57d26a052d804b81264f9ebca081f45

krudny commented:
We refetch on windows focus a lot of things. Do we have to? Maybe we could establish some limits

<img width="581" height="229" alt="image" src="https://github.com/user-attachments/assets/397a39ba-ca0f-4871-89a7-587fa77e7f6b" />

For example refetching requirements or variants or criterias every tab switch is useless. Criteria are not going to change at all. Variants as well. But if you change, then please make sure that all related things are invalidated after group creation / edition. We could also state some reasonable cache time, for example we can assume that targets didnt change for x seconds

krudny commented:
<img width="1151" height="909" alt="image" src="https://github.com/user-attachments/assets/34657e8d-ca30-4bfa-bd31-ccd9a87f6de4" />

nit: some scroll issue in variants (our fav)

krudny commented:
When i select student from course group x, then change filters to course group y then my project group selection is cleared. Not super important, but i was confused. 

I got more confused when sometimes i could select more course groups, sometimes not. As a user, i would like to have information about allow cross course group value. Because we might have 2 projects in one course, one would allow cross course grouping, one not, it might be confusing for instructor why he cannot select more that 1 course group

krudny commented:
I feel that student view is not ready at all. 
Current account is not a part of a project group, but that should be just two 200 requests not 8 of 404. 

<img width="276" height="207" alt="image" src="https://github.com/user-attachments/assets/8d7ecc22-f9e5-4459-be39-833ea9afb11b" />

Also modals on student view have different error sizes.

And i think that student part was mocked a long time ago and now noone changed anything. I am just wondering if all those hooks and service methods are used? Next question, are they used properly without logic duplication?

krudny commented:
In my mind, after refactoring to match architecture design, shared directory should be deleted. 

Filters defaults should be placed in filters components. Project modals should be in components/project/modals

krudny commented:
please update hooks to match architecture design

kligarski commented:
<details>
<summary>I will.</summary>

![aewzyi](https://github.com/user-attachments/assets/b89e7c1a-b9c1-417d-9715-745fdcbddd65)

</details>

kligarski committed:
fix imports after rebase

kligarski committed:
move new hooks to match new file structure

kligarski committed:
rwd

kligarski committed:
add stale time, fix typo

kligarski committed:
add additional info to filters

kligarski committed:
remove shared directory as this is totally in the scope of this PR fr fr

kligarski commented:
> poor rwd
> <img alt="image" width="377" height="669" src="https://private-user-images.githubusercontent.com/134285490/526195186-8b41a091-abbd-4ac4-a6f0-cd8e96447f7c.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU4Nzc3NDUsIm5iZiI6MTc2NTg3NzQ0NSwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MTk1MTg2LThiNDFhMDkxLWFiYmQtNGFjNC1hNmYwLWNkOGU5NjQ0N2Y3Yy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxNlQwOTMwNDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NmE5NWVhZTJjNzUyY2Q2MDdkNjdlNzNmYTE4MDg5ZmVhMWI0ZjU2OGNiMzhmZGY1MzUwNzNhNjhhZDY0MTFkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.g6PS9fCW6LXZiRKRRWcFWylPBkkOVODd1jZ5iiSY0B8">

I couldn't reproduce this but I refactored the group picking modal, should be better now.

> > If the title gets longer, it'll be truncated with "...". Having the type at the beginning prevents that. But I can change it if you prefer having it at the end.
> 
> cant we shrink just category to always fit [] content?

No, this would require refactoring `AccordionSection` to accept component instead of the string as section title. I'm not sure if it would look great either. I can try to do that if you're really convinced that it will look better but not sure if this is critical when taking into account our time constraints.

> We refetch on windows focus a lot of things. Do we have to? Maybe we could establish some limits
> <img alt="image" width="581" height="229" src="https://private-user-images.githubusercontent.com/134285490/526205222-397a39ba-ca0f-4871-89a7-587fa77e7f6b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU4Nzc3NzksIm5iZiI6MTc2NTg3NzQ3OSwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MjA1MjIyLTM5N2EzOWJhLWNhMGYtNDg3MS04OWE3LTU4N2ZhNzdlN2Y2Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxNlQwOTMxMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yM2M0YjZlZGJmYjBmYmQxZTk4MDAwOTY4ZjU5MTJkMmE3YjMzZThkMGZhYjc3N2E0NzViMTkwYTJmNGE3YjUxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.DEvnUgq0wbHEUJ0M_LS0J21bLmXyEYp9ZXstcPwmQsw">
> 
> For example refetching requirements or variants or criterias every tab switch is useless. Criteria are not going to change at all. Variants as well. But if you change, then please make sure that all related things are invalidated after group creation / edition. We could also state some reasonable cache time, for example we can assume that targets didnt change for x seconds

I added stale time.

> <img alt="image" width="1151" height="909" src="https://private-user-images.githubusercontent.com/134285490/526205596-34657e8d-ca30-4bfa-bd31-ccd9a87f6de4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU4Nzc3NzksIm5iZiI6MTc2NTg3NzQ3OSwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MjA1NTk2LTM0NjU3ZThkLWNhMzAtNGJmYS1iZDMxLWNjZDlhODdmNmRlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxNlQwOTMxMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02MmJkMmMxMWQ1MTlkMjAwN2Y4NGIxMTMwY2ExYjRhOTNhYjJjYWNlMjkzOWM4MmI2M2Q4NzZjN2Q1M2NlYTIwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.aBMosqpIkCovYuvktuC_82FRfUxHgq9pk7Il6jp19fM">
> 
> nit: some scroll issue in variants (our fav)

Refactoring column layout is out of scope of this PR. Either way it's usable, it just looks bad.

> When i select student from course group x, then change filters to course group y then my project group selection is cleared. Not super important, but i was confused.
> 
> I got more confused when sometimes i could select more course groups, sometimes not. As a user, i would like to have information about allow cross course group value. Because we might have 2 projects in one course, one would allow cross course grouping, one not, it might be confusing for instructor why he cannot select more that 1 course group

I added an additional info field in filters. Appropriate message needs to be supplied by the backend.

> I feel that student view is not ready at all. Current account is not a part of a project group, but that should be just two 200 requests not 8 of 404.
> <img alt="image" width="276" height="207" src="https://private-user-images.githubusercontent.com/134285490/526206656-8d7ecc22-f9e5-4459-be39-833ea9afb11b.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjU4Nzc3NzksIm5iZiI6MTc2NTg3NzQ3OSwicGF0aCI6Ii8xMzQyODU0OTAvNTI2MjA2NjU2LThkN2VjYzIyLWY5ZTUtNDQ1OS1iZTM5LTgzM2VhOWFmYjExYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTIxNlQwOTMxMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZWU5MDcyMTNlZGFlOTFjNTU3ODdmZTYyYjg3MjA1OWJkY2E5M2Q1MzYyNTVkMWZiOGE1MDAzNDU3OTc2M2JkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sfv3p_mx6ciT98OFk_ky_FiZ1ETt-AD7wFTT4D6NAmc">
> 
> Also modals on student view have different error sizes.
> 
> And i think that student part was mocked a long time ago and now noone changed anything. I am just wondering if all those hooks and service methods are used? Next question, are they used properly without logic duplication?

This needs to be handled by the backend - empty arrays should be returned instead of an error. I fixed error message size.

> In my mind, after refactoring to match architecture design, shared directory should be deleted.
> 
> Filters defaults should be placed in filters components. Project modals should be in components/project/modals

Done.

> please update hooks to match architecture design

Done.

kligarski committed:
fix key in target list

kligarski committed:
fix error messages

kligarski committed:
rwd adjustment

kligarski committed:
minor adjustment

kligarski committed:
minor adjustment to minor adjustment

kligarski committed:
hide project-related speed dial items if project group not assigned

kligarski committed:
Merge branch 'develop' into feature/OG-126-projects-frontend

krudny commented on polymorphia-frontend/components/speed-dial/modals/group-info/student-info/StudentInfo.tsx:
```diff
@@ -0,0 +1,52 @@
+      </div>
+    );
+  }
+  if (isError || data === undefined) {
```
!data?

kligarski replied:
https://github.com/krudny/Polymorphia/pull/248/commits/6d12957a266bc2bcb4d6af1908d72c4be3282cd8

krudny commented on polymorphia-frontend/components/speed-dial/modals/group-info/student-info/StudentInfo.tsx:
```diff
@@ -0,0 +1,52 @@
+
+  return (
+    <div className="flex flex-col gap-2 min-w-80">
+      {data?.map((student) => {
```
after 3 ifs we are still not sure if data exists? i would also rename it to more descriptive one. Also inline styles

kligarski replied:
https://github.com/krudny/Polymorphia/pull/248/commits/6d12957a266bc2bcb4d6af1908d72c4be3282cd8

krudny commented on polymorphia-frontend/components/speed-dial/modals/project-group-configuration/group-pick/index.css:
```diff
@@ -0,0 +1,39 @@
+}
+
+.group-pick-filter-btn {
+  @apply !mx-0 !py-0 !border-0 !border-b-2 !rounded-none shrink-0;
```
this is our technical debt, but i know that root code is mine, so no worries

krudny commented on polymorphia-frontend/components/speed-dial/modals/project-group-configuration/variant-pick/index.tsx:
```diff
@@ -0,0 +1,223 @@
+  };
+
+  const sectionIds = new Set(
+    projectCategories.map((category) => String(category.id))
```
i see this many times, it could be accordion generic util, but not important at all

kligarski replied:
Decided to leave it for now.

krudny reviewed (approved):
thank you Krzysztof, leftovers are under backend pr but we can merge this and local storage filters to not block things for longer

kligarski committed:
minor cleanup

kligarski committed:
student info scrollability
